pubtools/pulplib/_impl/client/client.py

```diff
@@ -181,6 +181,15 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def get_maintenance_status(self):
+        """Get maintenance status from maintenance repository
+
+        Returns:
+            Future[dict]
```

Comment:
Could you please make this a proper object (attrs-based class, perhaps a PulpObject subclass) with a documented and enforced interface?

I don't want to be returning raw dicts here, it means anyone using this API needs to somehow know what's in the dict and how to interpret it.  It should rather be the job of this library to interpret the raw data from Pulp.

You should probably also add a new schema for the file.  As far as I know, nobody wrote a proper reference for the format of this file yet.  This library seems a reasonable place for it to live.